riff-networking/src/main/java/com/wepay/riff/network/NetworkClient.java

```diff
@@ -114,18 +114,22 @@ public void closeAsync() {
             if (isValid()) {
                 state.set(ClientState.CLOSING);
 
-                channelFuture.addListener(f -> {
-                    if (f.isSuccess()) {
-                        Channel channel = ((ChannelFuture) f).channel();
-                        if (channel.isOpen()) {
-                            channel.close().addListener(f2 -> shutdown());
-                        } else {
-                            shutdown();
+                if (channelFuture != null) {
+                    channelFuture.addListener(f -> {
+                        if (f.isSuccess()) {
+                            Channel channel = ((ChannelFuture) f).channel();
+                            if (channel.isOpen()) {
+                                channel.close().addListener(f2 -> shutdown());
+                            } else {
+                                shutdown();
+                            }
                         }
-                    }
-                });
+                    });
+                    return;
+                }
             }
         }
+        shutdown();
```

Comment:
`shutdown()` shouldn't be called when `isValid()` is false.

Reply:
fixed.